### PR TITLE
Minor caching changes and readme update

### PR DESCRIPTION
**Pull request by sutrkiller**
### Motivation

Minor fixes in caching.

### Checklist

- [x] Code follows coding conventions held in this repo
- [x] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

Automatic tests

### CODE REVIEW

**Comment by sutrkiller** [comment]
Do not merge yet.

File: README.md [hunk 1]
```diff
@@ -86,13 +86,15 @@ Rich text elements in Kentico Kontent can contain links to other content items.
...
+If you need updated content as soon as possible, you can use another caching strategy where entries are invalidated by webhooks. In order to enable this strategy just switch to another implemented [caching client](https://github.com/Kentico/kontent-boilerplate-net/blob/master/src/content/Kentico.Kontent.Boilerplate/Caching/Webhooks/CachingDeliveryClient.cs) by using `IServiceCollection.AddWebhookInvalidatedCachingClient` extension in [Startup](https://github.com/Kentico/kontent-boilerplate-net/blob/6fb2b26deecb858f3853d84e29121e3f17b3a291/src/content/Kentico.Kontent.Boilerplate/Startup.cs#L53). Also, you need to create a webhook.
+
+To create the webhook, go to Project settings --> Webhooks --> Create new Webhook. Give it a name (like "Cache invalidation webhook") and the publicly routable URL of your app with `/Webhooks/KenticoKontent` appended (like "https://myapp.azurewebsites.net/Webhooks/KenticoKontent"). Then, copy the API secret and paste it as the `KenticoKontentWebhookSecret` environment variable (secret) into your app's settings (e.g. appsettings.json).
```
Comment (petrsvihlik) [inline README.md]: it's not `/Webhooks/KenticoKontent` but `/Webhooks/Webhooks`
image

Comment (sutrkiller) [reply README.md]: Updated

File: README.md [hunk 2]
```diff
@@ -86,13 +86,15 @@ Rich text elements in Kentico Kontent can contain links to other content items.
 
 ### How to set up webhook-enabled caching
```
Comment (alesk-kentico) [inline README.md]: 20 seconds is good for demonstration purposes, but it is not good for caching in a production environment. Please consider changing this value to 10 minutes.

File: README.md [hunk 3]
```diff
@@ -86,13 +86,15 @@ Rich text elements in Kentico Kontent can contain links to other content items.
...
 
-To create the web hook, go to Project settings --> Webhooks --> Create new Webhook. Give it a name (like "Cache invalidation webhook") and the publicly routable URL of your app with `/WebHooks/KenticoKontent` appended (like "https://myapp.azurewebsites.net/WebHooks/KenticoKontent"). Then, copy the API secret and paste it as the `KenticoKontentWebhookSecret` environment variable (secret) into your app's settings.
+If you need updated content as soon as possible, you can use another caching strategy where entries are invalidated by webhooks. In order to enable this strategy just switch to another implemented [caching client](https://github.com/Kentico/kontent-boilerplate-net/blob/master/src/content/Kentico.Kontent.Boilerplate/Caching/Webhooks/CachingDeliveryClient.cs) by using `IServiceCollection.AddWebhookInvalidatedCachingClient` extension in [Startup](https://github.com/Kentico/kontent-boilerplate-net/blob/6fb2b26deecb858f3853d84e29121e3f17b3a291/src/content/Kentico.Kontent.Boilerplate/Startup.cs#L53). Also, you need to create a webhook.
```
Comment (alesk-kentico) [inline README.md]:
```suggestion
If displaying outdated content for a limited time is not an option, you need to use another caching strategy and invalidate cached content when a webhook notification about content change is received. To enable this caching strategy just switch to another [caching client](https://github.com/Kentico/kontent-boilerplate-net/blob/master/src/content/Kentico.Kontent.Boilerplate/Caching/Webhooks/CachingDeliveryClient.cs) by using `IServiceCollection.AddWebhookInvalidatedCachingClient` extension in [Startup](https://github.com/Kentico/kontent-boilerplate-net/blob/6fb2b26deecb858f3853d84e29121e3f17b3a291/src/content/Kentico.Kontent.Boilerplate/Startup.cs#L53). Also, you need to create a webhook.
```

File: README.md [hunk 4]
```diff
@@ -86,13 +86,15 @@ Rich text elements in Kentico Kontent can contain links to other content items.
...
 
-![New webhook configuration](https://i.imgur.com/ootVcPZ.png)
+To create the webhook, go to Project settings --> Webhooks --> Create new Webhook. Give it a name (like "Cache invalidation webhook") and the publicly routable URL of your app with `/Webhooks/Webhooks` appended (like "https://myapp.azurewebsites.net/Webhooks/Webhooks"). Then, copy the API secret and paste it as the `KenticoKontentWebhookSecret` environment variable (secret) into your app's settings (e.g. appsettings.json).
```
Comment (alesk-kentico) [inline README.md]:
```suggestion
Also, you need to [create a webhook](https://docs.kontent.ai/tutorials/develop-apps/integrate/using-webhooks-for-automatic-updates#a-creating-a-webhook). When entering a webhook URL, append a `/Webhooks/Webhooks` path to a publicly available URL address of the application, e.g. `https://myapp.azurewebsites.net/Webhooks/Webhooks`. Finally, copy the API secret and put it into the app settings (usually the appsettings.json) as the `KenticoKontentWebhookSecret` environment variable.
```

File: README.md (hunk 3 above)
Comment (alesk-kentico) [inline README.md]:
```suggestion
If displaying outdated content for a limited time is not an option, you need to use another caching strategy and invalidate cached content when a webhook notification about content change is received. To enable this caching strategy just switch to another [caching client](https://github.com/Kentico/kontent-boilerplate-net/blob/master/src/content/Kentico.Kontent.Boilerplate/Caching/Webhooks/CachingDeliveryClient.cs) by using `IServiceCollection.AddWebhookInvalidatedCachingClient` extension in [Startup](https://github.com/Kentico/kontent-boilerplate-net/blob/6fb2b26deecb858f3853d84e29121e3f17b3a291/src/content/Kentico.Kontent.Boilerplate/Startup.cs#L53).
```

**Comment by xantari** [comment]
Why is items feed caching removed?

**Comment by alesk-kentico** [comment]
> Why is items feed caching removed?

The items feed endpoint is intended for specific use cases, e.g. building a static site or exporting content. These use cases usually require lots of of content that is up to date and there's no point in caching it. Are you aware of a use case when caching items feed responses might improve application performance and resilience?

**Comment by xantari** [comment]
No use cases that I can think of, just wondering.

**Comment by sutrkiller** [comment]
Alright, you can probably merge it @petrsvihlik if you think it's ok.

File: src/content/Kentico.Kontent.Boilerplate/Areas/WebHooks/Controllers/WebhooksController.cs [hunk 5]
```diff
@@ -46,7 +46,7 @@ public IActionResult Index([FromBody] WebhookModel model)
                     dependencies.Add(CacheHelper.GetTypesDependencyKey());
                 }
 
-                if (model.Message.Type == "content_type")
+                if (model.Message.Type.Equals("content_type", StringComparison.Ordinal))
```
Comment (petrsvihlik) [inline src/content/Kentico.Kontent.Boilerplate/Areas/WebHooks/Controllers/WebhooksController.cs]: why equals?

File: src/content/Kentico.Kontent.Boilerplate/Caching/CacheHelper.cs [hunk 6]
```diff
@@ -174,7 +174,7 @@ public static IEnumerable<string> GetItemJsonDependencies(JObject response)
 
             return dependencies.Count > MAX_DEPENDENCY_ITEMS
                 ? new[] { GetItemsDependencyKey() }
-                : dependencies.Distinct();
```
Comment (petrsvihlik) [inline src/content/Kentico.Kontent.Boilerplate/Caching/CacheHelper.cs]: why this change?

File: src/content/Kentico.Kontent.Boilerplate/Caching/CacheHelper.cs [hunk 7]
```diff
@@ -334,10 +324,10 @@ private static bool IsItemListingResponse(dynamic response)
 
         private static bool IsComponent(JProperty property)
         {
-            // Components have 0 on index 14 in its id.
+            // Components have substring 01 in its id starting on position 14.
             // xxxxxxxx-xxxx-0xxx-xxxx-xxxxxxxxxxxx
```
Comment (petrsvihlik) [inline src/content/Kentico.Kontent.Boilerplate/Caching/CacheHelper.cs]: the comment should be updated i guess

File: test/Kentico.Kontent.Boilerplate.Tests/Caching/ResponseHelper.cs [hunk 8]
```diff
@@ -80,7 +80,7 @@ public static class ResponseHelper
         internal static (string codename, object item) CreateComponent()
         {
             var id = Guid.NewGuid().ToString();
-            id = $"{id.Substring(0, 14)}0{id.Substring(15)}";
+            id = $"{id.Substring(0, 14)}01{id.Substring(16)}";
```
Comment (petrsvihlik) [inline test/Kentico.Kontent.Boilerplate.Tests/Caching/ResponseHelper.cs]: I'd appreciate a comment similar to one above (with "XXX...")...at least. :)

File: src/content/Kentico.Kontent.Boilerplate/Areas/WebHooks/Controllers/WebhooksController.cs (hunk 5 above)
Comment (sutrkiller) [reply src/content/Kentico.Kontent.Boilerplate/Areas/WebHooks/Controllers/WebhooksController.cs]: Yeah, this change doesn't make any sense as it's equivalent.

File: src/content/Kentico.Kontent.Boilerplate/Caching/CacheHelper.cs (hunk 6 above)
Comment (sutrkiller) [reply src/content/Kentico.Kontent.Boilerplate/Caching/CacheHelper.cs]: Dependencies variable is of type HashSet so no need to call distinct.

Comment (petrsvihlik) [reply src/content/Kentico.Kontent.Boilerplate/Caching/CacheHelper.cs]: true :)

File: src/content/Kentico.Kontent.Boilerplate/Areas/WebHooks/Controllers/WebhooksController.cs (hunk 5 above)
Comment (petrsvihlik) [reply src/content/Kentico.Kontent.Boilerplate/Areas/WebHooks/Controllers/WebhooksController.cs]: i'd keep it simple then :D
